### PR TITLE
fakeroot 1.32

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  In file included from libfakeroot_time64.c:19:
  ./wrapped.h:43:88: error: unknown type name 'posix_spawn_file_actions_t'
  extern int my_posix_spawn (pid_t * __restrict pid, const char * __restrict path, const posix_spawn_file_actions_t *file_actions, const posix_spawnattr_t * __restrict attrp, char *const argv[ __restrict], char *const envp[ __restrict]) __attribute__((visibility("hidden")));
                                                                                         ^
  ./wrapped.h:43:136: error: unknown type name 'posix_spawnattr_t'
  extern int my_posix_spawn (pid_t * __restrict pid, const char * __restrict path, const posix_spawn_file_actions_t *file_actions, const posix_spawnattr_t * __restrict attrp, char *const argv[ __restrict], char *const envp[ __restrict]) __attribute__((visibility("hidden")));
                                                                                                                                         ^
  ./wrapped.h:44:89: error: unknown type name 'posix_spawn_file_actions_t'
  extern int my_posix_spawnp (pid_t * __restrict pid, const char * __restrict path, const posix_spawn_file_actions_t *file_actions, const posix_spawnattr_t * __restrict attrp, char *const argv[ __restrict], char *const envp[ __restrict]) __attribute__((visibility("hidden")));
                                                                                          ^
  ./wrapped.h:44:137: error: unknown type name 'posix_spawnattr_t'
  extern int my_posix_spawnp (pid_t * __restrict pid, const char * __restrict path, const posix_spawn_file_actions_t *file_actions, const posix_spawnattr_t * __restrict attrp, char *const argv[ __restrict], char *const envp[ __restrict]) __attribute__((visibility("hidden")));
                                                                                                                                          ^
  4 errors generated.
```

**Comment by carlocab** [comment]
@Homebrew/core anyone here have a Debian Salsa account and can upstream the embedded patch? I registered for one but it seems to take ages to get confirmed.

File: Formula/fakeroot.rb [hunk 1]
```diff
@@ -51,3 +54,17 @@ def install
...
+--- a/wrapawk_macosx
++++ b/wrapawk_macosx
+@@ -10,6 +10,7 @@ BEGIN{
+   print WARNING > headerfile;
+   print "#ifndef WRAPPED_H" > headerfile;
+   print "#define WRAPPED_H" > headerfile;
++  print "#include <spawn.h>" > headerfile;
+   print "#define MY_GLUE2(a,b) a ## b"      > headerfile;
+   print "#define MY_DEF(a) MY_GLUE2(my_,a)" > headerfile;
+   print WARNING > deffile;
```
Comment (ZhongRuoyu) [inline Formula/fakeroot.rb:70]: 🙋 I have an activated Salsa account.

By the way, would this be more appropriate:

```diff
diff --git a/wrapfunc.inp b/wrapfunc.inp
index f592aad..fa2ff40 100644
--- a/wrapfunc.inp
+++ b/wrapfunc.inp
@@ -48,10 +48,11 @@ fgetattrlist;int;(int fd, void *attrList, void *attrBuf, size_t attrBufSize, uns
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_5
 getattrlist$UNIX2003;int;(const char *path, void *attrList, void *attrBuf, size_t attrBufSize, unsigned long options);(path, attrList, attrBuf, attrBufSize, options)
 #endif
 #endif
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_5
+#include <spawn.h>
 lstat$INODE64;int;(const char *file_name, struct stat *buf);(file_name, buf)
 stat$INODE64;int;(const char *file_name, struct stat *buf);(file_name, buf)
 fstat$INODE64;int;(int fd, struct stat *buf);(fd, buf)
 posix_spawn;int;(pid_t * __restrict pid, const char * __restrict path, const posix_spawn_file_actions_t *file_actions, const posix_spawnattr_t * __restrict attrp, char *const argv[ __restrict], char *const envp[ __restrict]);(pid, path, file_actions, attrp, argv, envp)
 posix_spawnp;int;(pid_t * __restrict pid, const char * __restrict path, const posix_spawn_file_actions_t *file_actions, const posix_spawnattr_t * __restrict attrp, char *const argv[ __restrict], char *const envp[ __restrict]);(pid, path, file_actions, attrp, argv, envp)
```

Tested locally; seems to work.

Comment (carlocab) [reply Formula/fakeroot.rb:70]: Yea, that looks better.

**Comment by ZhongRuoyu** [comment]
> anyone here have a Debian Salsa account and can upstream the embedded patch?

https://salsa.debian.org/clint/fakeroot/-/merge_requests/27

**Review by carlocab — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**